homeassistant/components/light/yeelight.py

```diff
@@ -286,13 +328,428 @@ def set_flash(self, flash) -> None:
+
+            if effect == EFFECT_CHRISTMAS:
+                count = 0
+                duration = 300
+
+                transitions = list()
+                transitions.append(
+                    RGBTransition(255, 0, 0, brightness=self.brightness,
+                                  duration=duration))
+                transitions.append(SleepTransition(duration=3000))
+                transitions.append(
+                    RGBTransition(0, 255, 0, brightness=self.brightness,
+                                  duration=duration))
+                transitions.append(SleepTransition(duration=3000))
+                flow = Flow(count=count, transitions=transitions)
+
+            if effect == EFFECT_RGB:
+                count = 0
+                duration = 300
+
+                transitions = list()
+                transitions.append(
+                    RGBTransition(255, 0, 0, brightness=self.brightness,
+                                  duration=duration))
+                transitions.append(SleepTransition(duration=3000))
+                transitions.append(
+                    RGBTransition(0, 255, 0, brightness=self.brightness,
+                                  duration=duration))
+                transitions.append(SleepTransition(duration=3000))
+                transitions.append(
+                    RGBTransition(0, 0, 255, brightness=self.brightness,
+                                  duration=duration))
+                transitions.append(SleepTransition(duration=3000))
+                flow = Flow(count=count, transitions=transitions)
+
+            if effect == EFFECT_RGB_TRANS:
+                count = 0
+                duration = 10000
+
+                transitions = list()
+                transitions.append(
+                    RGBTransition(255, 0, 0, brightness=self.brightness,
+                                  duration=duration))
+                transitions.append(
+                    RGBTransition(0, 255, 0, brightness=self.brightness,
+                                  duration=duration))
+                transitions.append(
+                    RGBTransition(0, 0, 255, brightness=self.brightness,
+                                  duration=duration))
+                flow = Flow(count=count, transitions=transitions)
+
+            if effect == EFFECT_ENJOY:
+                count = 0
+                duration = 40000
+
+                transitions = list()
+                transitions.append(
+                    TemperatureTransition(1700, duration=duration))
+                transitions.append(
+                    TemperatureTransition(6500, duration=duration))
+                flow = Flow(count=count, transitions=transitions)
+
+            if effect == EFFECT_RANDOM_LOOP:
+                count = 0
+                duration = 3000
+
+                transitions = list()
+                transitions.append(
+                    RGBTransition(random.randrange(0, 255),
```

Comment:
`random.randrange(0, 255)` is used in the flux component. i did't give it any thought. So i should change it into `radom.randint(0, 255)` right?